include/joypad.h

```diff
@@ -0,0 +1,427 @@
+    JOYPAD_ACCESSORY_TYPE_RUMBLE_PAK,
+    /** @brief Transfer Pak accessory. */
+    JOYPAD_ACCESSORY_TYPE_TRANSFER_PAK,
+    /** @brief Bio Sensor accessory. */
+    JOYPAD_ACCESSORY_TYPE_BIO_SENSOR,
+    /** @brief Pokemon Snap Station accessory. */
+    JOYPAD_ACCESSORY_TYPE_SNAP_STATION,
+} joypad_accessory_type_t;
+
+/** @brief Joypad Buttons */
+typedef union joypad_buttons_u
+{
+    /** @brief Raw button data as a 16-bit value. */
+    uint16_t raw;
+    /// @cond
+    struct __attribute__((packed))
+    {
+    /// @endcond
+        /** @brief State of the A button */
+        unsigned a : 1;
+        /** @brief State of the B button */
+        unsigned b : 1;
+        /** @brief State of the Z button */
+        unsigned z : 1;
+        /** @brief State of the Start button */
+        unsigned start : 1;
+        /** @brief State of the D-Pad Up button */
+        unsigned d_up : 1;
+        /** @brief State of the D-Pad Down button */
+        unsigned d_down : 1;
+        /** @brief State of the D-Pad Left button */
+        unsigned d_left : 1;
+        /** @brief State of the D-Pad Right button */
+        unsigned d_right : 1;
+        /**
+         * @brief State of the Y button.
+         * 
+         * This input only exists on GameCube controllers.
+         */
+        unsigned y : 1;
+        /**
+         * @brief State of the X button.
+         * 
+         * This input only exists on GameCube controllers.
+         */
+        unsigned x : 1;
+        /** @brief State of the digital L trigger */
+        unsigned l : 1;
+        /** @brief State of the digital R trigger */
+        unsigned r : 1;
+        /**
+         * @brief State of the C-Up button.
+         * 
+         * For GameCube controllers, the value will be
+         * emulated based on the C-Stick Y axis position.
+         */
+        unsigned c_up : 1;
+        /**
+         * @brief State of the C-Down button.
+         * 
+         * For GameCube controllers, the value will be
+         * emulated based on the C-Stick Y axis position.
+         */
+        unsigned c_down : 1;
+        /**
+         * @brief State of the C-Left button.
+         * 
+         * For GameCube controllers, the value will be
+         * emulated based on the C-Stick X axis position.
+         */
+        unsigned c_left : 1;
+        /**
+         * @brief State of the C-Right button.
+         * 
+         * For GameCube controllers, the value will be
+         * emulated based on the C-Stick X axis position.
+         */
+        unsigned c_right : 1;
+    /// @cond
+    };
+    /// @endcond
+} joypad_buttons_t;
+
+/** @brief Joypad Inputs Unified State Structure */
+typedef struct __attribute__((packed)) joypad_inputs_s
+{
+    /// @cond
+    union
+    {
+    /// @endcond
+        /** @brief Structure containing the state of all buttons. */
+        joypad_buttons_t buttons;
+        /// @cond
+        struct __attribute__((packed))
+        {
```

Comment:
It's just syntactic sugar. I wanted `joypad_inputs_t` to offer a flat namespace for convenience (so that devs could write `inputs.a` instead of forcing them to use `inputs.btn.a`) but unfortunately that means copy-pasting the content of `joypad_buttons_t` inline here.

I'd be interested to gather some wider user feedback on this, but my suspicion was that they would prefer to type fewer characters. It's not a huge burden on our side for a small quality-of-life enhancement.

Reply:
OK. On the other hand, do we need `buttons` instead? I think you need it internally for implementing the pressed/released functions, but do we need it exposed? Could this be a private, double underscored, undocumented members instead?

I agree with your statement of a small burden on our side to improve user's life. What I'm just a bit concerned of is documenting two different fields that contain the same value. That seems confusing on the user side.

Reply:
Changed `buttons` substruct to a private double-underscored undocumented member in dc2f2cae8cf9a852928b022cb25d1353cf09e72d

I originally exposed it publicly simply because I was using it internally and thought it might be useful, but I don't see any reason why a dev would need it considering `joypad_read_n64_inputs_sync` is no longer public and `joypad_get_buttons` exists.